grouper/permissions.py

```diff
@@ -63,7 +63,8 @@ def grant_permission(session, group_id, permission_id, argument=''):
     Throws:
         AssertError if argument does not match ARGUMENT_VALIDATION regex
     """
-    assert re.match(ARGUMENT_VALIDATION, argument), 'Permission argument does not match regex.'
+    assert re.match(ARGUMENT_VALIDATION + r"$", argument), \
```

Comment:
These should probably have `r"^"` prepended as well, right?

Reply:
`re.match` only matches at the start of the string (as opposed to `re.search`). It's one of the weird things about Python regexes.